carla_ego_vehicle/src/carla_ego_vehicle/carla_ego_vehicle.py

```diff
@@ -350,32 +392,32 @@ def run(self):
         """
         main loop
         """
-        # wait for ros-bridge to set up CARLA world
-        rospy.loginfo("Waiting for CARLA world (topic: /carla/world_info)...")
-        try:
-            rospy.wait_for_message("/carla/world_info", CarlaWorldInfo, timeout=10.0)
-        except rospy.ROSException as e:
-            rospy.logerr("Timeout while waiting for world info!")
-            raise e
-        rospy.loginfo("CARLA world available. Spawn ego vehicle...")
+        if ROS_VERSION == 1:
```

Comment:
We do not wait for ROS2 ? We miss an elif?

Reply:
ROS2's `rclpy` unfortunately doesn't have any simple function to create a subsciber, wait for one message, and then unsubscribe.